test/actions/FilterActionsTest.js

```diff
@@ -0,0 +1,180 @@
+            const newHistory = []
+            filterActions.getUrlFromFilter(newHistory, filterNoEventTypes)
+            expect(newHistory).to.be.empty
+        })
+    })
+
+    describe('generateUrl', function() {
+        const history = []
+        const filter = {ticketId: 0, eventTypes: [1, 2]}
+
+        const result = filterActions.generateUrl(history, filter)
+
+        const expectedValue = '/tickets/0?eventTypes=1&eventTypes=2'
+
+        it('should return the right URL from the filter object', function() {
+            expect(result).to.equal(expectedValue)
+        })
+    })
+
+    describe('findEventTypeInRef', function() {
+        const testEventTypeId = 0
+        const unknownEventTypeId = 10000
+
+        const referenceData = {0: {id: 0, name: 'testZero'}, 1: {id: 1, name: 'testOne'}}
+
+        const expectedEventType = { id: 0, name: 'testZero' }
+        const unknownEventType = {id: 10000, name: 'unknown'}
+
+        it('should return a matching object when given a known id', function() {
+            const result = filterActions.findEventTypeInRef(testEventTypeId, referenceData)
+            expect(result).to.deep.equal(expectedEventType)
+        })
+        it('should return a new object with name "unknown" when given an unknown id', function() {
+            const badResult = filterActions.findEventTypeInRef(unknownEventTypeId, referenceData)
+            expect(badResult).to.deep.equal(unknownEventType)
+        })
+    })
+
+    describe('changeEventFilter', function() {
+        const history = []
+        const filter = {ticketId: 0, eventTypes: [1,2]}
+
+        const result = filterActions.changeEventFilter(history)(filter)
+
+        const expectedValue = {type: 'CHANGE_EVENT_FILTER', filter: {ticketId: 0, eventTypes: [1, 2]}}
+        
+        it('should return an object with a type and a filter', function() {
+            expect(result).to.deep.equal(expectedValue)
+        })
+    })
+
+    describe('applyFilter', function() {
+        const history = []
+        const oldFilter = {incidentId: 0}
+        const goodNewFilter = {incidentId: 0}
+        const badNewFilter = {foo: 'bar'}
+
+        const dispatch = function(arg) {
+            return arg
+        }
+        const errorMessage = 'Need to filter on incidentId!'
+        const expectedReturn = { type: 'CHANGE_EVENT_FILTER', filter: {incidentId: 0} }
+
+        it('should throw an error when new filter has no incident id', function() {
+            expect(() => filterActions.applyFilter(history)(oldFilter, badNewFilter)(dispatch)).to.throw(errorMessage)
+        })
+    })
+
+    describe('serializeEventTypesForQuery', function() {
+        const eventTypes = [1, 2]
+        const noEventTypes = []
+
+        const expectedResult = 'eventTypes=1&eventTypes=2'
+
+        const result = filterActions.serializeEventTypesForQuery(eventTypes)
+
+        it('should return a serialized array', function() {
```

Comment:
The expected return value is a string of query parameters, and query params are a method of serializing dictionaries/hashes. Let's specify 'string of query params'. BTW, I dove down a research hole on query params and discovered the conventional syntax is `'eventTypes[]=1&eventTypes[]=2'`, though this appears to be a framework dependent thing.

Reply:
That started in PHP and is not guaranteed to work in other places. The Moment.js team once had a really long discussion about it for gosh-knows-what-reason.